10.detect-separation/scripts/nbconverted/1.separate.py

```diff
@@ -0,0 +1,324 @@
+
+# coding: utf-8
+
+# # Detect Separation Between Two Phenotypes in TCGA, GTEx, and TARGET Data
+# 
+# **Gregory Way, 2019**
+# 
+# Perform a t-test to isolate the specific k dimension and algorithm that best distinguishes the two groups.
+
+# In[1]:
+
+
+import os
+import sys
+import numpy as np
+import pandas as pd
+from scipy.stats import ttest_ind, ttest_rel
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+sys.path.append("../8.gtex-interpret")
+from scripts.utils import (
+    load_weight_matrix,
+    apply_signature,
+    load_enrichment_results,
+    extract_feature,
+)
+
+sys.path.append("../9.tcga-classify/")
+from scripts.tcga_util import build_feature_dictionary
+
+
+# In[2]:
+
+
+def ttest_sex_difference(feature_series, male_ids, female_ids):
+    """
+    To be applied to a pandas dataframe by column
+    """
+    feature_name = feature_series.name
+    feature_algorithm, feature_num = feature_name.split('_')
+    
+    male_activation = feature_series[feature_series.index.isin(male_ids)]
+    female_activation = feature_series[feature_series.index.isin(female_ids)]
+    
+    # Perform t-test on two groups
+    t_stat, t_p = ttest_ind(male_activation, female_activation)
+    
+    return([t_stat, t_p, feature_algorithm, feature_num])
+
+
+# ## 1. GTEx Sex Analysis
+
+# In[3]:
+
+
+# Load GTEx phenotype data
+file = os.path.join("..", "0.expression-download", "download", "GTEx_v7_Annotations_SubjectPhenotypesDS.txt")
+gtex_pheno_df = pd.read_table(file)
+gtex_pheno_df.head()
+
+
+# In[4]:
+
+
+gtex_z_matrix_dict = build_feature_dictionary(dataset="GTEX",
+                                              load_data=True,
+                                              store_train_test='test')
+
+
+# In[5]:
+
+
+# Extract male and female ids from the dataset
+example_matrix_df = gtex_z_matrix_dict['signal']['8']['451283']['test']
```

Comment:
Is this for testing?

Reply:
I am just extracting the IDs in this example matrix. I will update the comment to be more clear, thanks!